src/react-queries/queries/common.ts

```diff
@@ -90,43 +89,15 @@ export const common = createQueryKeys('common', {
   }),
   evmFees: ({
     data,
-    wallet,
     accountNumber,
     to,
-    from,
     value,
-    feeAsset,
-    feeAssetMarketData,
-  }: {
-    to: string
-    // Only used to make the queryKey unique by from address, since wallet doesn't serialize and is excluded from the queryKey
-    from: string
-    accountNumber: number
-    wallet: HDWallet
-    data: string
-    value: string
-    feeAsset: Asset
-    feeAssetMarketData: MarketData
+    chainId,
+  }: Omit<GetFeesWithWalletArgs, 'wallet' | 'adapter' | 'accountNumber' | 'data'> & {
+    accountNumber: number | undefined
+    data: string | undefined
+    chainId: ChainId
   }) => ({
-    queryKey: ['evmFees', to, from, accountNumber, data, value],
-    queryFn: async () => {
-      const adapter = assertGetEvmChainAdapter(fromAssetId(feeAsset.assetId).chainId)
-
-      const fees = await getFeesWithWallet({
-        adapter,
-        data,
-        wallet,
-        to,
-        value,
-        accountNumber,
-      })
-
-      const txFeeFiat = bn(fromBaseUnit(fees.networkFeeCryptoBaseUnit, feeAsset.precision))
-        .times(feeAssetMarketData.price)
-        .toString()
-
-      const { networkFeeCryptoBaseUnit } = fees
-      return { fees, txFeeFiat, networkFeeCryptoBaseUnit }
-    },
```

Comment:
And now the queryFn (in `useEvmFees`) is only concerned about calling `getFeesWithWallet` - the adapter bit is handled by the hook, and the bits after are handled with a `select` fn to derive it, in the spirit of keeping queryFn as close to their inner async fn as possible, not trying to do too much and doing derivy bits in `select`